### PR TITLE
Handle null values in MetaData.mergeMaps, preventing potential NPE

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal
Prevent NPE crash when attempting to deliver error reports.

## Design

`MetaData` tab values can be `HashMaps`, whose entries can be null. `MetaData.mergeMaps` seemed to assume that the `Map` was always a `ConcurrentHashMap`, whose entries cannot be null. 

On React Native this manifested as a NPE when merging `config.metaData` and `report.metaData` due to `app.activeScreen` being intermittently recorded as null. This can occur on any platform which uses bugsnag-android, however.

## Changeset

Added a null check when adding the base value to the merged map.

## Tests

Wrote unit test + mazerunner scenario, and manually verified that valid reports are sent from a React Native example app using a local bugsnag-android artefact.

## Review

For the submitter, initial self-review:

- [ ] Commented on code changes inline explain the reasoning behind the approach
- [ ] Reviewed the test cases added for completeness and possible points for discussion
- [ ] A changelog entry was added for the goal of this pull request
- [ ] Check the scope of the changeset - is everything in the diff required for the pull request?
- This pull request is ready for:
  - [ ] Initial review of the intended approach, not yet feature complete
  - [ ] Structural review of the classes, functions, and properties modified
  - [ ] Final review

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency across platforms for structures or concepts added or modified
- [ ] Consistency between the changeset and the goal stated above
- [ ] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [ ] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [ ] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

File: sdk/src/main/java/com/bugsnag/android/MetaData.java [hunk 1]
```diff
@@ -146,9 +146,7 @@ static MetaData merge(@NonNull MetaData... metaDataList) {
                 Object overridesValue = map.get(key);
 
                 if (overridesValue != null) {
-                    if (baseValue != null
-                        && baseValue instanceof Map
-                        && overridesValue instanceof Map) {
+                    if (baseValue instanceof Map && overridesValue instanceof Map) {
```
Comment (fractalwrench) [inline sdk/src/main/java/com/bugsnag/android/MetaData.java]: `baseValue != null` is redundant when also using `instanceof`

**Comment by fractalwrench** [comment]
This was originally discovered when setting the codeBundleId in React Native, which in turn sets metadata on the native layer: https://github.com/bugsnag/bugsnag-react-native/issues/291

File: sdk/src/androidTest/java/com/bugsnag/android/MetaDataMergeTest.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+    @Test
+    public void testConcurrentMapMerge() {
+        MetaData metaData = new MetaData();
+
+        Map<Object, Object> nestedMap = new HashMap<>();
+        metaData.addToTab("foo", "bar", nestedMap);
+        nestedMap.put("whoops", null);
+
+        Map<String, Object> mergedMap = MetaData.merge(metaData, metaData).store;
```
[the remark this answers is not in the document]
Comment (fractalwrench) [reply sdk/src/androidTest/java/com/bugsnag/android/MetaDataMergeTest.java]: I have updated the test to generate two different MetaData objects.

**Comment by fractalwrench** [comment]
@snmaynard I believe I have addressed the comments, are you happy with the updates?

**Comment by snmaynard** [comment]
I pinged you with my concerns - we can discuss tomorrow!
